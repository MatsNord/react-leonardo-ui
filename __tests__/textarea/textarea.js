import React from 'react';
import renderer from 'react-test-renderer';

import Textarea from '../../src/textarea';

test('Normal textarea renders correctly', () => {
  const tree = renderer.create(
    <Textarea value="Value" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Textarea with states renders correctly', () => {
  const tree = renderer.create(
    <Textarea value="Value" invalid disabled />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Textarea with states renders correctly', () => {
  const tree = renderer.create(
    <Textarea value="Value" active disabled />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Textarea with modifiers renders correctly', () => {
  const tree = renderer.create(
    <Textarea value="Value" variant="inverse" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Textarea passes props correctly', () => {
  const onClick = jest.fn(() => {});

  const tree = renderer.create(
    <Textarea value="Value" className="my-class" data-x="x" title="Title" onClick={onClick} />
  ).toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onClick();
  expect(onClick.mock.calls.length).toBe(1);
});
