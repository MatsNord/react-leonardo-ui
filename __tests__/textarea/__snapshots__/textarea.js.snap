exports[`test Normal textarea renders correctly 1`] = `
<textarea
  className="lui-textarea"
  value="Value" />
`;

exports[`test Textarea passes props correctly 1`] = `
<textarea
  className="lui-textarea my-class"
  data-x="x"
  onClick={[Function]}
  title="Title"
  value="Value" />
`;

exports[`test Textarea with modifiers renders correctly 1`] = `
<textarea
  className="lui-textarea lui-textarea--inverse"
  value="Value" />
`;

exports[`test Textarea with states renders correctly 1`] = `
<textarea
  className="lui-textarea lui-invalid lui-disabled"
  value="Value" />
`;

exports[`test Textarea with states renders correctly 2`] = `
<textarea
  active={true}
  className="lui-textarea lui-disabled"
  value="Value" />
`;
